### PR TITLE
[sil-cleanup] Clean up the pass and only invalidate analyzes if we ac…

### PR DESCRIPTION
**Pull request by gottesmm**
…tually change something.

We also only need to invalidate the instructions analysis for sil-cleanup.

rdar://39335800

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@atrick This is just a simple cleanup of the pass in preparation for my adding the special code for the semantics tag.

File: lib/SILOptimizer/Transforms/SILCleanup.cpp [hunk 1]
```diff
@@ -9,13 +9,18 @@
...
+/// \file
+///
+/// Cleanup SIL to make it suitable for IRGen.
+///
+/// We perform the following canonicalizations:
+///
+/// 1. We remove calls to Builtin.staticReport(), which are not needed post SIL.
+///
+/// FIXME: This pass is mandatory so should probably be in
+/// SILOptimizer/Mandatory.
```
Comment (slavapestov) [inline lib/SILOptimizer/Transforms/SILCleanup.cpp]: Why not just move it to the mandatory directory

Comment (gottesmm) [reply lib/SILOptimizer/Transforms/SILCleanup.cpp]: I have a chain of patches and I don't want to run into merge issues. I wish I had started it there. But I think git is smart enough now that I can do the move afterwards without messing with the git blame too much.

Comment (gottesmm) [reply lib/SILOptimizer/Transforms/SILCleanup.cpp]: But you know what, maybe git's merge is good enough for this to "commute" the merge with the move.

**Comment by gottesmm** [comment]
I am going to open a new PR with the moved thing.

**Pull request closed without merging**